cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -159,15 +159,22 @@ private Publication performUpdate(PublicationRepresentations publicationRepresen
         return attempt(() -> getExistingPublication(publicationRepresentation))
                    .map(publicationRepresentation::withExistingPublication)
                    .map(PublicationUpdater::update)
-                   .map(publicationRepresentations -> publicationRepresentations.updateHasEffectiveChanges()
-                                                          ? update(publicationRepresentations)
-                                                          : publicationRepresentations.getExistingPublication())
+                   .map(this::getPublication)
                    .orElseThrow();
     }
 
+    private Publication getPublication(PublicationRepresentations publicationRepresentations) {
+        if (publicationRepresentations.updateHasEffectiveChanges()) {
```

Comment:
the "ifNeeded" in the else indicate that there is some logic for unneeded that may be missing (and I can't see)

Reply:
I wanted to indicate that we do not persist nvi report for "all" entries, but only those which satisfy some conditions, i.e. are nvi reported publications.
private void persistNviReportIfNeeded(PublicationRepresentations publicationRepresentations) {
        if (hasScientificResource(publicationRepresentations)) {
            var nviReport = NviReport.fromPublicationRepresentation(publicationRepresentations);
            saveNviReport(publicationRepresentations, nviReport);
        }
    }